test/conformance/helpers/broker_tracing_test_helper.go

```diff
@@ -70,274 +69,171 @@ func BrokerTracingTestHelper(t *testing.T, channel metav1.TypeMeta, setupClient
-		Span: tracinghelper.MatchHTTPClientSpanNoReply(filterHost, loggerTriggerPath),
-		Children: []tracinghelper.TestSpanTree{
-			{
-				Note: "18. Broker Filter for the 'logger' trigger receives the event from the Broker TrChannel.",
-				Span: tracinghelper.MatchHTTPServerSpanNoReply(filterHost, loggerTriggerPath),
-				Children: []tracinghelper.TestSpanTree{
-					{
-						Note: "19. Broker Filter for the 'logger' trigger sends the event to the logger pod.",
-						Span: tracinghelper.MatchHTTPClientSpanNoReply(loggerSVCHost, "/"),
-						Children: []tracinghelper.TestSpanTree{
-							{
-								Note: "20. Logger pod receives the event from the Broker Filter for the 'logger' trigger.",
-								Span: tracinghelper.MatchHTTPServerSpanNoReply(loggerSVCHost, "/"),
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-
-	// Steps 13-20. Directly steps 15-16. 17-20 are included as children.
-	loggerEventIngressToTrigger := tracinghelper.TestSpanTree{
-		Note: "13. Broker Ingress sends the event to the Broker's TrChannel.",
-		Span: tracinghelper.MatchHTTPClientSpanNoReply(triggerChanHost, "/"),
-		Children: []tracinghelper.TestSpanTree{
-			{
-				Note: "14. Broker TrChannel receives the event from the Broker Ingress.",
-				Span: tracinghelper.MatchHTTPServerSpanNoReply(triggerChanHost, "/"),
-				Children: []tracinghelper.TestSpanTree{
-					// Steps 15-16.
-					loggerEventSentFromTrChannelToTransformer,
-					// Steps 17-20.
-					loggerEventSentFromTrChannelToLogger,
-				},
-			},
-		},
-	}
-
-	// Steps 7-10: Event from TrChannel sent to transformer Trigger and its reply to the InChannel.
-	transformerEventSentFromTrChannelToTransformer := tracinghelper.TestSpanTree{
-		Note: "7. Broker TrChannel sends the event to the Broker Filter for the 'transformer' trigger.",
-		Span: tracinghelper.MatchHTTPClientSpanNoReply(filterHost, transformerTriggerPath),
-		Children: []tracinghelper.TestSpanTree{
-			{
-				Note: "8. Broker Filter for the 'transformer' trigger receives the event from the Broker TrChannel.",
-				Span: tracinghelper.MatchHTTPServerSpanNoReply(filterHost, transformerTriggerPath),
-				Children: []tracinghelper.TestSpanTree{
-					{
-						Note: "9. Broker Filter for the 'transformer' trigger sends the event to the transformer pod.",
-						Span: tracinghelper.MatchHTTPClientSpanWithReply(transformerSVCHost, "/"),
-						Children: []tracinghelper.TestSpanTree{
-							{
-								Note: "10. Transformer pod receives the event from the Broker Filter for the 'transformer' trigger.",
-								Span: tracinghelper.MatchHTTPServerSpanWithReply(transformerSVCHost, "/"),
-							},
-						},
-					},
-				},
+	return func(
+		t *testing.T,
+		channel *metav1.TypeMeta,
+		client *lib.Client,
+		loggerPodName string,
+		tc TracingTestCase,
+	) (tracinghelper.TestSpanTree, lib.EventMatchFunc) {
+		// Create the Broker.
+		client.CreateConfigMapPropagationOrFail(defaultCMPName)
```

Comment:
This should also move in the if block right below. Otherwise, it will block forever since MT broker won't reconcile ConfigMapPropagation and it never becomes ready.

Reply:
Yeah, so all the MT broker tests are failing (well, the ones I looked at with):
```
           tracker.go:152: Waiting for all KResources to become ready
            operation.go:168: Failed to get all test resources ready: failed waiting for {TypeMeta:{Kind:ConfigMapPropagation APIVersion:configs.internal.knative.dev/v1alpha1} ObjectMeta:{Name:eventing GenerateName: Namespace:includes-incoming-trace-id-dspdl SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[]}} to become ready: timed out waiting for the condition
```

Reply:
After making this change I can locally run the tests without that message.

```
vaikas-a01:eventing vaikas$ git diff
diff --git a/test/conformance/helpers/broker_tracing_test_helper.go b/test/conformance/helpers/broker_tracing_test_helper.go
index 6ca6e200..dc17ff2b 100644
--- a/test/conformance/helpers/broker_tracing_test_helper.go
+++ b/test/conformance/helpers/broker_tracing_test_helper.go
@@ -83,8 +83,9 @@ func setupBrokerTracing(brokerClass string) SetupInfrastructureFunc {
                tc TracingTestCase,
        ) (tracinghelper.TestSpanTree, lib.EventMatchFunc) {
                // Create the Broker.
-               client.CreateConfigMapPropagationOrFail(defaultCMPName)
                if brokerClass == eventing.ChannelBrokerClassValue {
+                       // create ConfigMapPropagation required by the Broker
+                       client.CreateConfigMapPropagationOrFail(defaultCMPName)
                        // create required RBAC resources including ServiceAccounts and ClusterRoleBindings for Brokers
                        client.CreateRBACResourcesForBrokers()
                }
```


However, I'm still seeing this failure:

```
            channel_tracing_test_helper.go:158: FAIL: No traceid in 1 messages: ([{ Validation: valid
                Context Attributes,
                  specversion: 1.0
                  type: logger
                  source: sender
                  id: 4c556633-a938-4160-8cb5-0e15482fa75f
                  time: 2020-03-25T02:32:24.308319132Z
                  datacontenttype: application/json
                Extensions,
                  knativearrivaltime: 2020-03-25T02:32:24.374852436Z
                  knativehistory: br-kne-trigger-kn-channel.includes-incoming-trace-id-hc82p.svc.cluster.local; br-kne-trigger-kn-channel.includes-incoming-trace-id-hc82p.svc.cluster.local
                  traceparent: 00-bbb409d58f273670d2af8be42a8782d2-df3fca54f39ffe6d-01
                Data,
                  "{\"msg\":\"TestBrokerTracing 4c556633-a938-4160-8cb5-0e15482fa75f\",\"sequence\":\"1\"}"
                 map[Accept-Encoding:[gzip] Ce-Knativearrivaltime:[2020-03-25T02:32:24.374852436Z] Ce-Knativehistory:[br-kne-trigger-kn-channel.includes-incoming-trace-id-hc82p.svc.cluster.local; br-kne-trigger-kn-channel.includes-incoming-trace-id-hc82p.svc.cluster.local] Ce-Traceparent:[00-bbb409d58f273670d2af8be42a8782d2-df3fca54f39ffe6d-01] Content-Length:[89] User-Agent:[Go-http-client/1.1] X-B3-Sampled:[1] X-B3-Spanid:[c8515f9e6674be5a] X-B3-Traceid:[bbb409d58f273670d2af8be42a8782d2]]}])
```

Reply:
My bad, the issue was that PR #2780 did not include mtbroker since it was authored before mtbroker was merged, but I hadn't yet pulled PR #2780 into my local branch. Fixed by enabling W3C trace propagation format in mtbroker components.

Reply:
For some reason I'm not hitting an issue with the config map propagation, but I'm not sure why.